apps/process_tpstream.cxx

```diff
@@ -264,69 +280,21 @@ int main(int argc, char const *argv[])
       //
       // TA Processing
       //
-      // Create the output buffer
-      std::vector<triggeralgs::TriggerActivity> ta_buffer;
-
-      // Loop over TPs
-      size_t n_tas = 0;
 
-      for( const auto& tp : tp_buffer ) {
-
-        if ( tp_filter(tp) ){
-          // if(!quiet)
-              // fmt::print("  TP filtered out!");
-          continue;
-        }
-
-        (*ta_maker)(tp, ta_buffer);
-
-        if (n_tas != ta_buffer.size()) {
-          if (!quiet) {
-            for( size_t i=n_tas; i<ta_buffer.size(); ++i){
-              const auto& ta = ta_buffer[i];
-              fmt::print("  + {} : ca_s={} ca_e={} ta_s={}, ta_e={} d_ta = {} | tp_s={} tp_s-ta_s={}\n", i, ta.channel_start, ta.channel_end, ta.time_start, ta.time_end, ta.time_end-ta.time_start, tp.time_start, (tp.time_start-ta.time_start) );
-            }
-          }
-          n_tas = ta_buffer.size();
-        }
-      }
+      const auto ta_start = std::chrono::steady_clock::now();
+      std::unique_ptr<daqdataformats::Fragment> ta_frag = ta_emulator->emulate(tp_buffer);
+      const auto ta_end = std::chrono::steady_clock::now();
```

Comment:
Ok, so this is a large change here, that changes the way the emulation works, and is different from how we thought about making latency measurements. The idea was to push tp into the tamaker one-by-one, and make latency measurements  based on the last TP entering TAMaker before making a TA.

Here the latency (given the `continue` below) will be the time the last TP bunch enters TAMaker and a (potentially bunch) of TAs made for these TPs. These measurements will break down if e.g. a bunch of TPs make 1 TA, and 98% of another TA -- they will look larger than what they are.

Would it be possible to change this to take TPs one-by-one instead, including the latency measurements below? It looks like an easy change anyway.